File: app/src/main/java/com/example/volleyrvpractice/FavouriteRecipeModel/FavouriteRecipeDao.java
package com.example.volleyrvpractice.FavouriteRecipeModel;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Observable;

@Dao
public interface FavouriteRecipeDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertFR(FavouriteRecipe fr);
    @Delete
    void deleteFR(FavouriteRecipe fr);
    @Query("Delete from favourite_recipe_table")
    void deleteAllFR();
    @Query("Select * from favourite_recipe_table")
    Observable<List<FavouriteRecipe>> getAllFR();
    @Query("Delete from favourite_recipe_table where primaryKey =:primaryKey")
    void deleteFRwithKey(int primaryKey);
}

File: app/src/main/java/com/jasonstudio/cookbook2/FavouriteRecipeModel/FavouriteRecipeDao.kt
package com.jasonstudio.cookbook2.FavouriteRecipeModel

import androidx.room.*
import io.reactivex.Observable

@Dao
interface FavouriteRecipeDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertFR(fr: FavouriteRecipe?)

    @Delete
    suspend fun deleteFR(fr: FavouriteRecipe?)

    @Query("Delete from favourite_recipe_table")
    suspend fun deleteAllFR()

    @get:Query("Select * from favourite_recipe_table")
    val allFR: Observable<List<FavouriteRecipe>>

    @Query("Delete from favourite_recipe_table where primaryKey =:primaryKey")
    suspend fun deleteFRwithKey(primaryKey: Int)
}